addons/website/models/website.py

```diff
@@ -769,17 +769,22 @@ def new_page(self, name=False, add_menu=False, template='website.default_page',
             page = self.env['website.page'].create(default_page_values)
             result['page_id'] = page.id
         if add_menu:
-            default_menu_values = {
-                'name': name,
-                'url': page_url,
-                'parent_id': website.menu_id.id,
-                'page_id': page.id,
-                'website_id': website.id,
-            }
-            if menu_values:
-                default_menu_values.update(menu_values)
-            menu = self.env['website.menu'].create(default_menu_values)
-            result['menu_id'] = menu.id
+            page_menu = self.env['website.menu'].search_count([
+                ('url', '=', page_url),
+                ('website_id', '=', website.id),
+            ], limit=1)
+            if not page_menu:
+                default_menu_values = {
+                    'name': name,
+                    'url': page_url,
+                    'parent_id': website.menu_id.id,
+                    'page_id': page.id,
+                    'website_id': website.id,
+                }
+                if menu_values:
+                    default_menu_values.update(menu_values)
+                menu = self.env['website.menu'].create(default_menu_values)
+                result['menu_id'] = menu.id
```

Comment:
Stability issue: now `'menu_id'` is not part of the result always, potentially breaking custo. In the end, it was already dubious to target stable for this fix, let's change our mind: I'll retarget to master.